mysql-compatibility: update the doc for v5.1 (#5760)
pingcap/docs: mysql-compatibility.md

In TiDB, all supported DDL changes are performed online. Compared with DDL operations in MySQL, the DDL operations in TiDB have the following major restrictions:

* Multiple operations cannot be completed in a single `ALTER TABLE` statement. For example, it is not possible to add multiple columns or indexes in a single statement. Otherwise, the `Unsupported multi schema change` error might be output.

> * Some type changings through `ALTER TABLE` statement is not supported. For example, TiDB does not support changing the field type from `DECIMAL` to `DATE`. Otherwise, the error information `Unsupported modify column: type %d not match origin %d` might be output. See [`ALTER TABLE`](/sql-statements/sql-statement-alter-table.md) for further details.

Reviewer: ```suggestion
* Some type changings through `ALTER TABLE` statement are not supported. For example, TiDB does not support changing the field type from `DECIMAL` to `DATE`. When a type change is not supported, the error  `Unsupported modify column: type %d not match origin %d` will be returned. See [`ALTER TABLE`](/sql-statements/sql-statement-alter-table.md) for further details.
```

Author: Thanks, for your suggestion, it has been addressed in @AilinKid 's https://github.com/pingcap/docs-cn/pull/6413.
